File: frontend/src/microservices/advertising/objects/AdvertisementReviewDecisionRequest.ts
// 本文件定义 advertising 模块的 `AdvertisementReviewDecisionRequest`，用于审核、驳回和下线展示等动作并提供 JSON 编解码。

export type AdvertisementReviewDecisionRequest = {
  advertisementId: string
  reviewerManagerId: string
  reviewNote: string | null
}

export const advertisementReviewDecisionRequestFromJson = (json: string): AdvertisementReviewDecisionRequest =>
  JSON.parse(json) as AdvertisementReviewDecisionRequest

export const advertisementReviewDecisionRequestToJson = (value: AdvertisementReviewDecisionRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/AdvertisementReviewDecisionRequest.scala
// 本文件定义 advertising 模块的 AdvertisementReviewDecisionRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AdvertisementReviewDecisionRequest(
    advertisementId: String,
    reviewerManagerId: String,
    reviewNote: Option[String]
)

object AdvertisementReviewDecisionRequest:
  given sourceEncoder: Encoder[AdvertisementReviewDecisionRequest] = deriveEncoder
  given sourceDecoder: Decoder[AdvertisementReviewDecisionRequest] = deriveDecoder

